meson.build

```diff
@@ -1,86 +1,177 @@
+      return 0;
+    }
+    ''',
+    name: 'gmtime_s can be used',
+  )
+    cargs += '-DHAVE_GMTIME_S=1'
+  endif
 endif
 
-if host_machine.system() == 'windows'
-  win = import('windows')
-  lcms2_srcs += win.compile_resources('Projects/VC2019/lcms2.rc')
-  cargs += '-DCMS_DLL_BUILD=1'
+jpeg_dep = dependency(
+  'libjpeg',
+  'jpeg',
+  required: get_option('jpeg'),
+)
+
+if jpeg_dep.found()
+  if (
+    cc.check_header('jconfig.h', dependencies: jpeg_dep)
+    and cc.check_header('jerror.h', dependencies: jpeg_dep)
+    and cc.check_header(
+      'jmorecfg.h',
+      prefix: '#include <jconfig.h>',
+      dependencies: jpeg_dep,
+    )
+    and cc.check_header(
+      'jpeglib.h',
+      prefix: '''
+    #include <stdio.h>
+    #include <stdlib.h>
+    ''',
+      dependencies: jpeg_dep,
+    )
+    and cc.compiles(
+      '''
+    #include <stdio.h>
+    #include <stdlib.h>
+    #include <jpeglib.h>
+    #if JPEG_LIB_VERSION < 62
+    #error IJG JPEG library must be version 6b or newer!
+    #endif
+    ''',
+      dependencies: jpeg_dep,
+      name: 'JPEG library is version 6b or later',
+    )
+  )
+    cargs += '-DHasJPEG=1'
+  else
+    message('JPEG library is not complete -- some components failed test')
+    jpeg_dep = disabler()
+  endif
+else
+  jpeg_dep = disabler()
 endif
 
-liblcms2_lib = library('lcms2', lcms2_srcs,
-  include_directories : inc_dirs,
-  gnu_symbol_visibility: 'hidden',
-  dependencies: deps,
-  c_args: cargs,
-  version: library_version,
-  vs_module_defs: 'src/lcms2.def',
-  install: true,
+tiff_dep = dependency(
+  'libtiff-4',
+  'libtiff',
+  'tiff',
+  required: get_option('tiff'),
 )
 
-liblcms2_dep = declare_dependency(
-  link_with : liblcms2_lib,
-  include_directories : inc_dirs
+if tiff_dep.found()
+  if (
+    cc.check_header('tiff.h', dependencies: tiff_dep)
+    and cc.check_header('tiffio.h', dependencies: tiff_dep)
+    and cc.check_header('tiffconf.h', dependencies: tiff_dep)
+    and cc.has_function('TIFFOpen', dependencies: tiff_dep)
+    and cc.has_function('TIFFClientOpen', dependencies: tiff_dep)
+    and cc.has_function('TIFFIsByteSwapped', dependencies: tiff_dep)
+  )
+    cargs += '-DHasTIFF=1'
+  else
+    message('TIFF library is not complete -- some components failed test')
+    tiff_dep = disabler()
+  endif
+else
+  tiff_dep = disabler()
+endif
+
+if (
+  not cc.compiles(
+    '''
+    #include <emmintrin.h>
+    int main() { __m128i n = _mm_set1_epi8(42); }
+  ''',
+    name: 'supports SSE2 intrinsics',
+  )
 )
+  cargs += '-DCMS_DONT_USE_SSE2=1'
+endif
+
+m_dep = cc.find_library('m', required: false)
+
+threads_dep = dependency('threads')
+
+if cc.has_function('pthread_mutex_lock', dependencies: threads_dep)
+  cargs += '-DHasTHREADS=1'
+else
+  cargs += '-DHasTHREADS=0'
+endif
+
+win = import('windows')
+
+deps = [m_dep, threads_dep]
+
+subdir('include')
+subdir('src')
+subdir('testbed')
+subdir('utils')
+
+if get_option('samples').enabled()
+  subdir('utils/samples')
+endif
```

Comment:
This is a tristate option, being misused to have two options mean the same thing. When configured with `-Dsamples=auto`, it is equivalent to disabling it entirely, when the semantic meaning is to check if required system dependencies and interfaces are available and it "can" be built.